File: app/src/main/java/ru/voronezhtsev/weatherapp/db/ForecastsRepository.java
package ru.voronezhtsev.weatherapp.db;

import ru.voronezhtsev.weatherapp.net.api.ForecastsService;
import ru.voronezhtsev.weatherapp.net.models.forecast.ForecastsResponse;
import rx.Single;

public class ForecastsRepository {
    private static final String MOSCOW = "524901";
    private static final String APP_ID = "458a017c6453d7ee6e2cfa3a5ddec547";

    private ForecastsDAO mForecastsDAO;
    private ResponseConverter mResponseConverter;
    private ForecastsService mForecastsService;

    public ForecastsRepository(ForecastsDAO forecastsDAO, ResponseConverter responseConverter,
                               ForecastsService forecastsService) {
        mForecastsDAO = forecastsDAO;
        mResponseConverter = responseConverter;
        mForecastsService = forecastsService;
    }

    public void save(ForecastsResponse response) {
        mForecastsDAO.addForecasts(mResponseConverter.convert(response));
    }

    public Single<ForecastsResponse> getForecast() {
        return mForecastsService.getForecasts(MOSCOW, APP_ID);
    }
}

File: app/src/main/java/ru/voronezhtsev/weatherapp/domain/ForecastsRepository.kt
package ru.voronezhtsev.weatherapp.domain

import io.reactivex.Single
import ru.voronezhtsev.weatherapp.models.domain.Forecast

interface ForecastsRepository {
    /**
     * Метод загружает данные о погоде из сети кладет их в БД и возвращает рез-т
     */
    fun loadFromNetwork(cityId: Long): Single<List<Forecast>>

    /**
     * Метод загружает данные о погоде из БД
     */
    fun loadFromDb(cityName: String): Single<List<Forecast>>
}